lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -871,12 +871,23 @@ def _handle_job_failure(self, job, job_state):
                 self.__cleanup_k8s_guest_ports(job_state.job_wrapper, job)
             self.__cleanup_k8s_job(job)
         except Exception:
-            log.exception("Could not clean up k8s batch job. Ignoring...")
+            log.exception("Could not clean up a failed k8s batch job. Ignoring...")
         return mark_failed
 
-    def __cleanup_k8s_job(self, job):
+    def __cleanup_k8s_job(self, job, retry=5, timeout=10):
+        log.debug(f"Cleaning up job with K8s id {job.name}.")
         k8s_cleanup_job = self.runner_params["k8s_cleanup_job"]
-        delete_job(job, k8s_cleanup_job)
+        for i in range(retry):
+            try:
+                delete_job(job, k8s_cleanup_job)
+                break
+            except Exception as exc:
```

Comment:
I would also narrow this to just the specific exception for which that makes sense.